llvm/lib/Analysis/InstructionSimplify.cpp

```diff
@@ -4316,12 +4309,30 @@ static Value *simplifyWithOpReplaced(Value *V, Value *Op, Value *RepOp,
   if (isa<FreezeInst>(I))
     return nullptr;
 
+  SmallVector<std::pair<Value *, Value *>> ValidReplacements{};
+  for (const auto &OpAndRepOp : Ops) {
+    // We cannot replace a constant, and shouldn't even try.
+    if (isa<Constant>(OpAndRepOp.first))
+      return nullptr;
```

Comment:
I realize this is pre-existing, but I do wonder why this is checked here and not *before* the trivial replacement at the top.

Reply:
Ill move.